src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,120 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.arrayContaining;
+
+class Exercise1 {
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortLocalComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new PersonComparator<>());
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    private static class PersonComparator<T extends Person> implements Comparator<T> {
+
+        @Override
+        public int compare(T firstPerson, T secondPerson) {
+            return Integer.compare(firstPerson.getAge(), secondPerson.getAge());
+        }
+    }
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person >() {
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                return Integer.compare(firstPerson.getAge(), secondPerson.getAge());
+            }
+        });
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                if(!firstPerson.getLastName().equals(secondPerson.getLastName())) {
+                    return firstPerson.getLastName().compareTo(secondPerson.getLastName());
+                } else {
+                    return firstPerson.getFirstName().compareTo(secondPerson.getFirstName());
+                }
+            }
+        });
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45),
+                new Person("Николай", "Зимов", 30),
+                new Person("Иван", "Мельников", 20)
+        )));
+    }
+
+    @Test
+    void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Predicate<Person> personPredicate = person -> ((Integer) 30).equals(person.getAge());
```

Comment:
You need to move here your implementation of `findFirstWithAge30UsingGuavaAnonymousPredicate()` method.

Reply:
done